Fix queue populator shared metrics

welansari opened the pull request:
- Queue Populator can be used for a single or multiple extensions at once. Currently the 'origin' label in the metrics is always set to "replication" which is not always true,  we change this to make it hold the list of loaded extensions in the current queue populator instance.
- Some queue populator metrics are shared between all extensions and not specific to replication. We change their prefix to `s3_backbeat_populator` instead of `s3_replication`

Issue: BB-527

Commit:
rename queue populator metrics

Some queue populator metrics are shared between all extensions and not
specific to replication.

Issue: BB-527

bert-e commented:
# Hello kerkesni,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://bert-e.scality.net/scality/backbeat/doc/user).


<details>
  <summary><b>Available options</b></summary>

  name   | description  | privileged   | authored
  ------ | ------------ | ------------ |----------
  `/after_pull_request` | Wait for the given pull request id to be merged before continuing with the current one. |  | 
  `/bypass_author_approval` | Bypass the pull request author's approval |  :star:  | 
  `/bypass_build_status` | Bypass the build and test status |  :star:  | 
  `/bypass_commit_size` | Bypass the check on the size of the changeset ```TBA``` |  :star:  | 
  `/bypass_incompatible_branch` | Bypass the check on the source branch prefix |  :star:  | 
  `/bypass_jira_check` | Bypass the Jira issue check |  :star:  | 
  `/bypass_peer_approval` | Bypass the pull request peers' approval |  :star:  | 
  `/bypass_leader_approval` | Bypass the pull request leaders' approval |  :star:  | 
  `/approve` | Instruct Bert-E that the author has approved the pull request. |  |  :writing_hand: 
  `/create_pull_requests` | Allow the creation of integration pull requests. |  | 
  `/create_integration_branches` | Allow the creation of integration branches. |  | 
  `/no_octopus` | Prevent Wall-E from doing any octopus merge and use multiple consecutive merge instead |  | 
  `/unanimity` | Change review acceptance criteria from `one reviewer at least` to `all reviewers` |  | 
  `/wait` | Instruct Bert-E not to run until further notice. |  | 
  

</details>



<details>
  <summary><b>Available commands</b></summary>

  name   | description  | privileged
  ------ | ------------ | ------------
  `/help` | Print Bert-E's manual in the pull request. | 
  `/status` | Print Bert-E's current status in the pull request ```TBA``` | 
  `/clear` | Remove all comments from Bert-E from the history ```TBA``` | 
  `/retry` | Re-start a fresh build ```TBA``` | 
  `/build` | Re-start a fresh build ```TBA``` | 
  `/force_reset` | Delete integration branches & pull requests, and restart merge process from the beginning. | 
  `/reset` | Try to remove integration branches unless there are commits on them which do not appear on the source branch. | 
  
</details>




*Status report is not available.*

bert-e commented:
# Incorrect fix version



The `Fix Version/s` in issue BB-527 contains:


* *None*



Considering where you are trying to merge, I ignored possible hotfix versions and I expected to find:



* `9.0.0`


Please check the `Fix Version/s` of BB-527, or the target
branch of this pull request.

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* 2 peers

Commit:
use proper label in queue populator metrics

Queue Populator can be used for a single or multiple
extensions at once. Currently the 'origin' label in the
metrics is always set to "replication" which is not always
true,  we change this to make it hold the list of loaded
extensions in the current queue populator instance.

Issue: BB-527

francoisferrand commented on .github/workflows/docker-build.yaml:
```diff
@@ -27,6 +27,13 @@ jobs:
           username: ${{ github.repository_owner }}
           password: ${{ github.token }}
 
+      - name: Install Oras
```
not needed, please rebase...

francoisferrand commented on lib/queuePopulator/BucketFileLogReader.js:
```diff
@@ -45,6 +46,7 @@ class BucketFileLogReader extends LogReader {
 
     getMetricLabels() {
         return {
+            origin: this._extensionNames,
```
should we not "join" the strings in the array? (here or maybe better in the constructor)

```this._extensionNames.join(',')```

welansari replied:
`_extensionNames ` is a string containing that. The `join` is done when setting up the log source in the queue populator

francoisferrand reviewed (approved):
it would be nice to document somewhere what "origin" really is : i.e. not the pod/process which generated the metric, but the kind of workflow/process from which the 'processing' was triggered... (i.e. so that we have replication count either from "replication" rules and "transition")

Commit:
document the "origin" label in queue populator metrics

Issue: BB-527

francoisferrand commented on lib/queuePopulator/QueuePopulator.js:
```diff
@@ -23,6 +23,10 @@ const {
+// origin is the name of the workflow that triggered the processing
+// of the log entries. The label can include multiple values when
+// the queue populator is shared between multiple workflows (e.g.
+// replication, lifecycle, notification).
```
nit:

```suggestion
// `replication,lifecycle,notification`).
```

welansari commented:
/approve

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/9.0`



The following branches have **NOT** changed:

* `development/7.10`
* `development/7.4`
* `development/7.70`
* `development/8.6`



Please check the status of the associated issue BB-527.

Goodbye kerkesni.




*The following options are set:* **approve**
